application/classes/Ushahidi/Repository/Tag.php

```diff
@@ -171,20 +171,21 @@ public function deleteTag($id)
 	 * @param $fullData
 	 * @return bool
 	 */
-	public function isRoleValid(Validation $validation, $fullData)
+	public function isRoleValid($tag)
 	{
 		$valid = true;
-		$entityFullData = $this->getEntity($fullData);
-		$isChild = !!$entityFullData->parent_id;
-		$hasRole = !!$entityFullData->role;
-		$parent = $isChild ? $this->selectOne(['id' => $entityFullData->parent_id]) : null;
-		if ($hasRole && $isChild && $parent) {
+		$isChild = !!$tag['parent_id'];
+		$parent = $isChild ? $this->selectOne(['id' => $tag['parent_id']]) : null;
+
+		// If tag has a role and is a child category
+		if ($isChild && $parent) {
```

Comment:
I've dropped `hasRole` because even if the tag *doesn't* have a role we should still double check that it matches the parent role (ie. no role on both!)

Reply:
I'll come back to this when my brain works properly ... but I think children tag usually didn't have a role setup from the client* and a null role in children usually represented just that it's the same as parent 
*the validation was for the weird cases where we somehow got one

Reply:
Ok. I'll check the flow. We should probably have set that role before this point if thats the case.

Reply:
It turns out this still fails, but also failed before. 
Basically: if the role is null the validator just ignore the result of this check entirely :( I'll figure out a workaround tomorrow

Reply:
Ugh So we didn't need to `hasRole` check. But we did have to manually set the error via `$validation` because Kohana sucks so much

Reply:
as long as categories always have a role, this will work . But "everyone can see" matches null role instead of a real role. That might be something we want to solve in the client ASAP with a 'annonymous' role, just saying that the hasRole check is needed at the moment.
source: checked in production and we don't send or receive a role for "Everyone can see" permission

Reply:
But thats sorta the point of my change. Previously: if we had a null role, the check didn't run so a parent could have role=admin, but the child would have role=null.
Don't we want to always force the role to match the parent? Or have a wildly misinterpreted?

TBH... the safer solution to all of this might be to forcibly overwrite the role in the UseCase

Reply:
>TBH... the safer solution to all of this might be to forcibly overwrite the role in the UseCase

yup.